packages/grafana-runtime/src/services/pluginExtensions/getPluginExtensions.ts

```diff
@@ -42,7 +43,7 @@ export type UsePluginComponentResult<Props = {}> = {
 };
 
 export type UsePluginComponentsResult<Props = {}> = {
-  components: Array<React.ComponentType<Props>>;
+  components: Array<React.ComponentType<Props> & { meta: PluginExtensionComponentMeta }>;
```

Comment:
nit: do we want a specific type for this instead?

Reply:
Yeah, I was thinking about that as well but was going a bit back and forth about it. The positive thing with having another type for this is that it is easier to reference in helper functions etc. The negative is that it might hide the fact that we are exposing the `React.ComponentType`.

Reply:
With that said, I'm happy to introduce the type you suggested.

Reply:
Ok, I understand. Then feel free to leave as is.